activerecord/lib/active_record.rb

```diff
@@ -274,6 +274,9 @@ def self.global_executor_concurrency # :nodoc:
   singleton_class.attr_accessor :belongs_to_required_validates_foreign_key
   self.belongs_to_required_validates_foreign_key = true
 
+  singleton_class.attr_accessor :before_committed_on_all_records
+  self.before_committed_on_all_records = false
```

Comment:
Note to self: if we defined this as a `class_variable` on `ActiveRecord::Base` we might be able to transition on a per class basis. 

Given how hairy this change can be it could be useful.

Reply:
Actually scratch that. The flag is checked in the transaction code, so doing a check on a per model basis would be super hairy.